lib/setupHooks/installCargoArtifactsHook.sh

```diff
@@ -55,6 +87,10 @@ prepareAndInstallCargoArtifactsDir() {
 
   case "${mode}" in
     "use-zstd")
+      compressAndInstallCargoArtifactsDirIncremental "${dir}" "${cargoTargetDir}"
+      ;;
+
+    "use-zstd-no-incr")
```

Comment:
I appreciate your attention to allowing one to opt into the previous behavior!

I'd okay to call out the new behavior (in the change log) and add a new environment variable (e.g. `doNotCompressIncrementally`) to `installCargoArtifactsHook.sh` which would give the old behavior back!

Reply:
I'd call it `useFullCargoArtifactsSnapshot`. I think in certain cases it might be useful to use it even when there are no problems with the incremental approach (just to shed the chain of incremental archives, etc.)

However I'm confused how to propagate it to this function. Can I just use a global variable?

BTW. When I starting documentation changes, it did occurred to me again that I think it's better to just use different `mode` string for it.

When it's a separate flag, than it's unclear when it's used, when ignored and it becomes hard to understand. When it is its own mode, then there's no problem like this, and it's clear then it's a slightly different mode.

I propose:

* `use-zstd-full`
* `use-zstd-incremental` or `use-zstd-diff`
* define the `use-zstd` and alias to "whatever we think is the best default", and make it `use-zstd-incremental` right now.

@ipetkov

Reply:
`installCargoArtifactsMode` allows us to only specify one thing, meaning if we want to subtly tweak different installation strategies we'd have to keep tacking on `use-zstd-flag-1`, `use-zstd-flag-2`, `use-zstd-flag-1-and-flag-2` which could get unweildy IMO

Reply:
We're not going to support that many variations, IMO. But if it's neccessary it could be

```
installCargoArtifactsMode = { mode = "use-zstd"; diff = true; };
```

which crane can easily detect and handle, I think.

@ipetkov